dataplane/src/main.rs

```diff
@@ -29,8 +30,23 @@ custom_target!("tonic", LevelFilter::ERROR, &[]);
 custom_target!("h2", LevelFilter::ERROR, &[]);
 custom_target!("Pyroscope", LevelFilter::INFO, &[]);
 
+fn init_name(args: &CmdArgs) {
+    if let Some(name) = args.get_name() {
+        set_name(name);
+    } else if let Ok(name) = gethostname() {
+        set_name(&name.to_string_lossy());
```

Comment:
Failure to convert the discovered hostname to utf8 cleanly is very much fatal IMO